src/utils/ApiExtract.ts

```diff
@@ -108,6 +113,66 @@ export class ApiExtractImpl implements ApiExtract {
 		return undefined;
 	};
 
+	/**
+	 * Checks whether a given aggregation is part of a UI5 symbol.
+	 * @example isAggregation("sap.m.NavContainer", "pages")
+	 * 	returns true
+	 * @example isAggregation("sap.m.App", "pages", false) // pages is a borrowed aggregation of "sap.m.NavContainer"
+	 * 	returns false
+	 */
+	isAggregation(symbolName: string, aggregationName: string, checkBorrowed = true): boolean {
+		return this.getAllOptionsByType(symbolName, "aggregation",
```

Comment:
Currently, the check for a specific aggregation first collects ALL aggregations incl. borrowed and only then checks the list. This could be improved by already checking for the specific aggregation during the collection (currently the while-loop in `getTypeByOption `), which would reduce unnecessary collection and parent checks if the aggregation is already found.

This also applies to the other `isXY` methods.